### PR TITLE
Adhere orchestrator zone APIs to new orchestrator contract

### PR DESCRIPTION
**Pull request by irinaepshteyn**
Adhere orchestrator zone APIs to new orchestrator contract [1]

[1] https://devcloud.swcoe.ge.com/devspace/pages/viewpage.action?spaceKey=PCE&title=GED+Services+-+Orchestrator+Integration#GEDServicesOrchestratorIntegration-1.GET

### CODE REVIEW

File: zone-service/src/main/java/org/cloudfoundry/identity/uaa/zone/OrchestratorZoneController.java [hunk 1]
```diff
@@ -1,16 +1,17 @@
 package org.cloudfoundry.identity.uaa.zone;
 
 import static org.cloudfoundry.identity.uaa.zone.ErrorMessageUtil.MANDATORY_VALIDATION_MESSAGE;
-import static org.cloudfoundry.identity.uaa.zone.ErrorMessageUtil.getErrorMessagesHttpMessageNotReadable;
 import static org.cloudfoundry.identity.uaa.zone.ErrorMessageUtil.getErrorMessagesConstraintViolation;
+import static org.cloudfoundry.identity.uaa.zone.ErrorMessageUtil.getErrorMessagesHttpMessageNotReadable;
 import static org.cloudfoundry.identity.uaa.zone.ErrorMessageUtil.getErrorMessagesMethodArgumentInvalid;
+import static org.cloudfoundry.identity.uaa.zone.OrchestratorState.CREATE_IN_PROGRESS;
```
Comment (irinaepshteyn) [inline zone-service/src/main/java/org/cloudfoundry/identity/uaa/zone/OrchestratorZoneController.java:7]: Let's not import static values for OrchestratorState.  Importing static values for FOUND and NOT_FOUND might be confused with HttpStatus.FOUND and HttpStatus.NOT_FOUND.

FYI, I removed all static imports for OrchestratorState in the next commit.

File: uaa/src/test/java/org/cloudfoundry/identity/uaa/mock/zones/OrchestratorZoneControllerMockMvcTests.java [hunk 2]
```diff
@@ -166,10 +166,12 @@ void testGetZone_Unauthorized(String url) throws Exception {
     void testGetZone_nameRequiredError(String url) throws Exception {
         performMockMvcCallAndAssertError(get(url), status().isBadRequest(),
                                          JsonUtils.writeValueAsString(
-                                             new OrchestratorErrorResponse("Required request parameter 'name' for " +
-                                                                           "method parameter type String is not " +
-                                                                           "present")),
-                orchestratorZonesReadToken);
+                                             new OrchestratorZoneResponse(null, null,
```
Comment (irinaepshteyn) [inline uaa/src/test/java/org/cloudfoundry/identity/uaa/mock/zones/OrchestratorZoneControllerMockMvcTests.java:169]: We could work toward gradually replacing performMockMvcCallAndAssertError with performMockMvcCallAndAssertResponse. There is no need to have both.

I changed it in the next commit.

File: uaa/src/test/java/org/cloudfoundry/identity/uaa/integration/OrchestratorZoneControllerIntegrationTests.java [hunk 3]
```diff
@@ -463,7 +464,7 @@ public void testCreateZone_ZoneAlreadyExists() {
         String errorMessage = String.format("The zone name %s is already taken. Please use a different " +
                                             "zone name", orchestratorZoneRequest.getName());
         assertEquals(JsonUtils.writeValueAsString(
-            new OrchestratorErrorResponse(errorMessage)), getResponseAlreadyExist.getBody());
+            new OrchestratorZoneResponse(zoneName, null, errorMessage, OrchestratorState.PERMANENT_FAILURE.toString())), getResponseAlreadyExist.getBody());
```
Comment (irinaepshteyn) [inline uaa/src/test/java/org/cloudfoundry/identity/uaa/integration/OrchestratorZoneControllerIntegrationTests.java:467]: It's better to compare OrchestratorZoneResponse than string is case serialization is implemented differently between controller and test.

I changed that in the next commit.

**Review by irinaepshteyn — COMMENTED** [review]
@johnson-jeb , if you agree with my comments, you could squash my next commit

File: zone-service/src/main/java/org/cloudfoundry/identity/uaa/zone/OrchestratorZoneController.java [hunk 4]
```diff
@@ -59,16 +61,18 @@ public ResponseEntity<OrchestratorZoneResponse> getZone(@NotBlank(message = MAND
 
     @PostMapping
     @Transactional
-    public ResponseEntity<?> createOrchestratorZone(@RequestBody @Valid OrchestratorZoneRequest orchestratorZoneRequest )
-        throws OrchestratorZoneServiceException, IOException {
+    public ResponseEntity<OrchestratorZoneResponse> createOrchestratorZone(
+        @RequestBody @Valid OrchestratorZoneRequest orchestratorZoneRequest) {
         zoneService.createZone(orchestratorZoneRequest);
-        return ResponseEntity.accepted().build();
+        OrchestratorZoneResponse body =
```
Comment (irinaepshteyn) [inline zone-service/src/main/java/org/cloudfoundry/identity/uaa/zone/OrchestratorZoneController.java:66]: In getZone(), OrchestratorZoneResponse is returned by the zoneService.getZoneDetails().  In createOrchestratorZone(), OrchestratorZoneResponse is built in the controller. For consistency we should use one or the other.

Comment (johnson-jeb) [reply zone-service/src/main/java/org/cloudfoundry/identity/uaa/zone/OrchestratorZoneController.java:66]: Sure I will move the code to inside the service

File: zone-service/src/main/java/org/cloudfoundry/identity/uaa/zone/OrchestratorZoneService.java [hunk 5]
```diff
@@ -206,6 +208,8 @@ public void createZone(OrchestratorZoneRequest zoneRequest) {
         } finally {
             IdentityZoneHolder.set(previous);
         }
+        return new OrchestratorZoneResponse(zoneRequest.getName(), null, ZONE_CREATED_MESSAGE,
```
[the remark this answers is not in the document]
Comment (johnson-jeb) [reply zone-service/src/main/java/org/cloudfoundry/identity/uaa/zone/OrchestratorZoneService.java]: Actually there is no state now in orchestrator like "zone created", so we are returning CREATE_IN_PROGRESS. But zone is created our side successfully. Orchestrator can read and understand state but end user can understand with message like zone is created in UAA.

File: zone-service/src/main/java/org/cloudfoundry/identity/uaa/zone/OrchestratorZoneService.java [hunk 6]
```diff
@@ -135,13 +135,14 @@ public void deleteZone(String zoneName) {
...
                                              IdentityZoneHolder.getCurrentZoneId()));
                 logger.debug("Zone - deleted id[" + zone.getId() + "]");
-                return;
             } else {
                 throw new OrchestratorZoneServiceException(zoneName, "Error : deleting zone Name[" + zoneName + "]");
             }
         } finally {
             IdentityZoneHolder.set(previous);
         }
+        return new OrchestratorZoneResponse(zoneName, null, ZONE_DELETED_MESSAGE,
```
[the remark this answers is not in the document]
Comment (johnson-jeb) [reply zone-service/src/main/java/org/cloudfoundry/identity/uaa/zone/OrchestratorZoneService.java]: Done.

**Review by Yogeshkumarb57 — APPROVED** [review]

**Pull request merged**